src/main/groovy/com/avioconsulting/mule/linter/model/MuleComponent.groovy

```diff
@@ -2,14 +2,21 @@ package com.avioconsulting.mule.linter.model
 
 class MuleComponent {
 
-    private String name
-    private Map<String, String> attributes = [:]
+    private final String componentName
+    private final String componentNamespace
+    private final Map<String, String> attributes = [:]
     private final List<MuleComponent> children
 
     MuleComponent(Map<String, String> attributes) {
         this(attributes, null)
     }
 
+    MuleComponent(String componentName, String componentNamespace, Map<String, String> attributes) {
+        this(attributes, null)
+        this.componentName = componentName
+        this.componentNamespace = componentNamespace
+    }
+
     MuleComponent(Map<String, String> attributes, List<MuleComponent> children) {
```

Comment:
See prior comment, whats the use case for creating a component without a name/namespace?

Reply:
removed this construction